crates/bevy_ecs/src/component.rs

```diff
@@ -241,6 +241,12 @@ impl ComponentInfo {
     fn new(id: ComponentId, descriptor: ComponentDescriptor) -> Self {
         ComponentInfo { id, descriptor }
     }
+
+    /// Create a new [`ComponentInfo`] for use in tests.
+    #[cfg(test)]
+    pub(crate) fn new_for_test(id: ComponentId, descriptor: ComponentDescriptor) -> Self {
```

Comment:
This does the same thing as `new`. It might be better just to make it `pub(crate)` and eschew the `#[cfg(test)]`.